portals/publisher/src/main/webapp/source/src/app/components/Apis/Details/NewOverview/AssociatedAPI.jsx

```diff
@@ -0,0 +1,161 @@
+                    if (!apiMap.has(key)) {
+                        apiMap.set(key, { apiName, apiVersion, apiId });
+                    }
+                }
+            }
+        });
+
+        return Array.from(apiMap.values());
+    };
+
+    const associatedAPIs = getAssociatedAPIs();
+
+    return (
+        <>
+            <div>
+                <Typography variant='h5' component='h3' className={parentClasses.title}>
+                    {associatedAPIs.length === 1 ? (
+                        <FormattedMessage
+                            id='Apis.Details.Overview.AssociatedAPI.title'
+                            defaultMessage='Associated API'
+                        />
+                    ) : (
+                        <FormattedMessage
+                            id='Apis.Details.Overview.AssociatedAPI.title'
+                            defaultMessage='Associated APIs'
+                        />
+                    )}
+                </Typography>
+            </div>
+            <Box p={1}>
+                <Grid container spacing={1}>
+                    {associatedAPIs.length > 0 ? (
+                        associatedAPIs.map((apiInfo, index) => (
+                            <React.Fragment key={`${apiInfo.apiName}-${apiInfo.apiVersion}`}>
+                                <Grid item xs={12} md={6} lg={4}>
+                                    <Typography component='p' variant='subtitle2' className={parentClasses.subtitle}>
+                                        <FormattedMessage
+                                            id='Apis.Details.Overview.AssociatedAPI.api.name'
+                                            defaultMessage='API Name {number}'
+                                            values={{ number: associatedAPIs.length === 1 ? '' : index + 1 }}
+                                        />
+                                    </Typography>
+                                </Grid>
+                                <Grid item xs={12} md={6} lg={8}>
+                                    <Typography component='p' variant='body1'>
+                                        {apiInfo.apiId ? (
+                                            <Link
+                                                to={getBasePath(API.CONSTS.API) + apiInfo.apiId + '/overview'}
+                                                target='_blank'
+                                                rel='noopener noreferrer'
+                                                style={{ 
+                                                    display: 'inline-flex', 
+                                                    alignItems: 'center', 
+                                                    textDecoration: 'none',
+                                                    color: 'inherit'
+                                                }}
+                                            >
+                                                {apiInfo.apiName}
```

Comment:
Shall we use the displayName here? The ids can have the apiName as that will be unique. We can change on the displayed name in the UI.